File: src/syncloud_platform/disks/path_checker.py
from os import path
from syncloudlib import logger


class PathChecker:

    def __init__(self, platform_config):
        self.platform_config = platform_config
        self.log = logger.get_logger('path_checker')

    def external_disk_link_exists(self):
        real_link_path = path.realpath(self.platform_config.get_disk_link())
        self.log.info('real link path: {0}'.format(real_link_path))

        external_disk_path = self.platform_config.get_external_disk_dir()
        self.log.info('external disk path: {0}'.format(external_disk_path))

        link_exists = real_link_path == external_disk_path
        self.log.info('link exists: {0}'.format(link_exists))

        return link_exists

File: backend/storage/path_checker.go
package storage

import (
	"github.com/syncloud/platform/config"
	"go.uber.org/zap"
	"path/filepath"
)

type PathChecker struct {
	config *config.SystemConfig
	logger *zap.Logger
}

type Checker interface {
	ExternalDiskLinkExists() bool
}

func NewPathChecker(config *config.SystemConfig, logger *zap.Logger) *PathChecker {
	return &PathChecker{
		config: config,
		logger: logger,
	}
}

func (c *PathChecker) ExternalDiskLinkExists() bool {
	realLinkPath, err := filepath.EvalSymlinks(c.config.DiskLink())
	if err != nil {
		c.logger.Error("cannot read disk link", zap.String("name", c.config.DiskLink()))
		return false
	}
	c.logger.Info("real link", zap.String("path", realLinkPath))

	externalDiskPath := c.config.ExternalDiskDir()
	c.logger.Info("external disk", zap.String("path", externalDiskPath))

	linkExists := realLinkPath == externalDiskPath
	c.logger.Info("link", zap.Bool("exists", linkExists))

	return linkExists
}
